include/pacemaker.h

```diff
@@ -186,6 +186,40 @@ int pcmk_list_nodes(xmlNodePtr *xml, char *node_types);
  */
 int pcmk_status(xmlNodePtr *xml);
 
+/*!
+ * \brief Check whether each rule in a list is in effect
+ *
+ * \param[in,out] xml       The destination for the result, as an XML tree
+ * \param[in]     input     The CIB XML to check (if \c NULL, use current CIB)
+ * \param[in]     date      Check whether the rule is in effect at this date and
+ *                          time (if \c NULL, use current date and time)
+ * \param[in]     rule_ids  The IDs of the rules to check, as a <tt>NULL</tt>-
+ *                          terminated list.
+ *
+ * \return Standard Pacemaker return code
+ */
+int pcmk_check_rules(xmlNodePtr *xml, xmlNodePtr input, crm_time_t *date,
+                     const char **rule_ids);
```

Comment:
This function could also be written to take `...` for the list of rules.  I'd have to look at how it's used elsewhere, but it could potentially save some work at the callers since you wouldn't have to construct the data type to pass.

Reply:
It would be easier for some callers so I like it in theory. However, I'm not aware of a way to convert an existing array of strings to a list of separate arguments (or even a `va_list`), and searching isn't turning anything useful up. Closest search result I found said you can't: https://stackoverflow.com/a/14705987/7660197. Let me know if you have ideas that don't require having a variadic version and a non-variadic version and we can give it a shot

I figure at least accepting a char* array is clear and straightforward, even if it requires a little bit of extra typing sometimes.

Reply:
There's no way that I know of to do that.

Reply:
Let me know; we can do it either way. I think there's a reasonable possibility that a caller will want to check multiple rules at once and parse the results together. Maybe fairly uncommon though...

I'd be less concerned about efficiency since it's probably an uncommon need and maybe not super time-sensitive; and more concerned about clunkiness for a caller. I've been annoyed at REST APIs that only let me get a single item at a time before.

Reply:
Might as well leave it as it is